server.go

```diff
@@ -53,27 +53,22 @@ looping:
 
 func (s *Server) readLoop(nc *NodeConn) error {
 	for {
-		_, msg, err := nc.Conn.WsConn.ReadMessage()
-		if err != nil {
-			return err
-		}
-
 		signal := signal{}
-		err = json.Unmarshal(msg, &signal)
+		err := nc.Conn.WsConn.ReadJSON(&signal)
 		if err != nil {
 			return err
 		}
 
-		err = s.notifyNode(signal.NodeID, msg)
+		err = s.notifyNode(&signal)
 		if err != nil {
 			return err
 		}
 	}
 }
 
-func (s *Server) notifyNode(nodeID discv5.NodeID, msg []byte) error {
-	if nc, ok := s.nodeChannels.Load(nodeID); ok {
-		(nc.(NodeConn)).writeChan <- &signal{Msg: msg}
+func (s *Server) notifyNode(signal *signal) error {
```

Comment:
Since we changed the function signature, maybe we should also rename the function?

How about `notify`?